### ISSUE TITLE
History Top Activity chart is misleading - spikey instead of solid!

### ISSUE THREAD

**Issue by giffenk** [akardapolov/ASH-Viewer#31]
I ran pgbench against a postgres database for 10 minutes. If I then later go back and view the history data, the "Top Activity" shown below does not present a solid chart, instead it shows a series of spikes.
I then selected the area between 2 spikes and the "Top sql & sessions" display does contain data.

So I claim the data is properly retained, but not displayed correctly in the Top Activity chart.
I will attach an image of a screen shot to illustrate the issue.

**Comment by giffenk** [comment]
![ashtopactivity](https://user-images.githubusercontent.com/30731876/74281535-9ab88c00-4cec-11ea-827b-f245a30879a3.png)

**Comment by giffenk** [comment]
If you select a longer period in the history then the Top activity is less spikey, but still misleading.

**Comment by giffenk** [comment]
![longerTopActivity](https://user-images.githubusercontent.com/30731876/74281991-66919b00-4ced-11ea-962e-8fb5d0ea7cd7.png)

**Comment by akardapolov** [comment]
Hi!

>> I ran pgbench against a postgres database for 10 minutes.

So, you have spikes (duration of 10 minutes) between launches. It's working well, as expected.

Thanks,
Alex.

**Comment by giffenk** [comment]
Agreed that it seems to be recording data properly. But I am having a very hard time convincing clients and co-workers that the data is correct. The visual graphs are misleading.

The expectation is that the visual graphs will remain consistent as you drill in or change the time span.

The lack of consistent visual representation of the same data leads the casual viewer to believe the data is also inconsistent.

The expectation is that a 10 minute solid chart in an hour view should also be shown as a solid 10 minute chart in a 10 minute view.

**Comment by giffenk** [comment]
When my clients viewed the first image I attached their immediate response was to ask why the system was peaking and then going to zero.

It was not. But the chart indicates it was.

**Comment by giffenk** [comment]
I understand that actual data is grabbed only once a second (and then averaged over a period?) but the expectation is that graphs / charts will blur the actual sampling interval and portray a continual trend.

**Comment by akardapolov** [comment]
Hi!

>> The expectation is that a 10 minute solid chart in an hour view should also be shown as a solid 10 minute chart in a 10 minute view.

Because the data in History module appeared using 15 minutes aggregation interval. So this is why you see spikes every 15 seconds in 10 minute view.

**Comment by akardapolov** [comment]
>> (and then averaged over a period?) 
1 second, 15 seconds and 1 minute

>> expectation is that graphs / charts will blur the actual sampling interval and portray a continual trend

It depends on data

**Comment by akardapolov** [comment]
https://github.com/akardapolov/ASH-Viewer/issues/31#issuecomment-584864800

In this case (you right) we need to interpolate data

Thanks,
Alex.
